docs/utilities.rst

```diff
@@ -0,0 +1,1421 @@
+Utilities
+-------------
+
+All functions can be imported directly from the ``eth_utils`` module
+
+Alternatively, you can get the curried version of the functions by
+importing them through the ``curried`` module like so:
+
+.. doctest::
+
+   >>> from eth_utils.curried import hexstr_if_str
+
+ABI Utils
+~~~~~~~~~
+
+``event_abi_to_log_topic(event_abi)`` -> bytes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Returns the 32 byte log topic for the given event abi.
+
+.. doctest::
+
+   >>> from eth_utils import event_abi_to_log_topic
+   >>> event_abi_to_log_topic({'type': 'event', 'anonymous': False, 'name': 'MyEvent', 'inputs': []})
+   b'M\xbf\xb6\x8bC\xdd\xdf\xa1+Q\xeb\xe9\x9a\xb8\xfd\xedb\x0f\x9a\n\xc21B\x87\x9aO\x19*\x1byR\xd2'
+
+``event_signature_to_log_topic(event_signature)`` -> bytes
```

Comment:
Not for this PR but we should look at migrating all of these into docstrings in the functions themselves and use `autodoc` to populate the docs.  Probably easier to maintain that way and our code should be easier to read as well.  I suspect the doctests could even be embedded.

Reply:
TIL about the pytest args for running code examples in docstrings:
https://github.com/ethereum/pyrlp/pull/97/files#diff-513c9b6861fc981fb2c708db477f542cL2

Reply:
Ahh cool, good to know thanks!!